### DISCUSSION TITLE
Can't seem to specify a title for an Enum field

### DISCUSSION THREAD

**Discussion by tlocke** [fastapi/fastapi#7165] [Questions]
When I try and specify a field title for an Enum type, the title doesn't appear in the docs. I'm new to FastAPI so I'm probably making a simple mistake. Here's an example:

```Python
from enum import Enum

from fastapi import FastAPI

from pydantic import BaseModel, Field

import uvicorn

app = FastAPI()


class ResolutionEnum(str, Enum):
    daily = 'daily'
    hourly = 'hourly'


class Item(BaseModel):
    resolution: ResolutionEnum = Field(title="Resolution of Item")


@app.post("/items/")
async def create_item(item: Item):
    return item


if __name__ == '__main__':
    uvicorn.run(app)
```

I'd expect the docs to give the title "Resolution of Item" but it just gives the default "ResolutionEnum". Thanks for your help!

**Comment by cahna** [comment]
Move the `title` to a docstring and the docs should look the way you want:

```
from enum import Enum

import uvicorn
from fastapi import FastAPI
from pydantic import BaseModel

app = FastAPI()


class ResolutionEnum(Enum):
    """Resolution of Item"""

    daily = "daily"
    hourly = "hourly"


class Item(BaseModel):
    resolution: ResolutionEnum


@app.post("/items/")
async def create_item(item: Item):
    return item


if __name__ == "__main__":
    uvicorn.run(app)
```

![image](https://user-images.githubusercontent.com/1045603/96613633-0a3b3e00-12b4-11eb-89d9-15909b836523.png)

**Reply by YuriiMotov** [reply]
Just one remark: this way you specify `decription`, not `title`

**Comment by tlocke** [comment]
Great, thanks for the quick response! Is this already documented somewhere, or shall I update the docs?

**Comment by cahna** [comment]
I am also new to FastAPI. I believe this has to do with pydantic and not FastAPI. The [pydantic docs seem to say that `Field(..., description="Resolution of Item")` should work](https://pydantic-docs.helpmanual.io/usage/schema/#field-customisation), but it doesn't for Enums. I have a habit of putting docstrings on everything, so I think that is how I learned it will work for pydantic+Enums.

**Comment by eseglem** [comment]
When you have it:

```python
class Item(BaseModel):
    resolution: ResolutionEnum = Field(title="Resolution of Item")
```

It has the `title` on `Item.resolution` but its an `Enum`, so none of the the `Field` logic is applied. See [Enums and Choices](https://pydantic-docs.helpmanual.io/usage/types/#enums-and-choices) for the lack of use of `Field`.

I cannot find a docs page that explicitly states `Field` can only be used on numerics and strings, but the [docstring](https://github.com/samuelcolvin/pydantic/blob/master/pydantic/fields.py#L142) seems to indicate that.

Even if it did work there, it would still only apply to `Item.resolution` and not the Enum. 

What you might want to do is use [schema customization](https://pydantic-docs.helpmanual.io/usage/schema/#schema-customization) to apply directly to the Enum. Or wherever you like.

**Comment by johnanthonyowens** [comment]
This has been fixed in pydantic 1.7. See https://github.com/samuelcolvin/pydantic/issues/1748

**Comment by tlocke** [comment]
Thanks @johnanthonyowens, I'll close this now.
